workers/data_refinery_workers/processors/test_compendia.py

```diff
@@ -325,16 +273,428 @@ def test_create_compendia_danio(self):
+                experiment,
+            )
+
+            micros.append(file)
+
+        experiment = Experiment()
+        experiment.accession_code = "GSE5678"
+        experiment.save()
+
+        result = ComputationalResult()
+        result.save()
+        rnas = []
+        for file in os.listdir("/home/user/data_store/raw/TEST/RNASEQ/"):
+
+            if "rnaseq.txt" in file:
+                continue
+
+            create_sample_for_experiment(
+                {
+                    "organism": danio_rerio,
+                    "accession_code": file,
+                    "technology": "RNA-SEQ",
+                    "filename": file,
+                    "data_dir": "/home/user/data_store/raw/TEST/RNASEQ/",
+                },
+                experiment,
+            )
+
+            rnas.append(file)
+
+        # Missing sample that will be filtered
+        sample = create_sample_for_experiment(
+            {
+                "organism": danio_rerio,
+                "accession_code": "GSM1487222",
+                "title": "this sample will be filtered",
+                "technology": "RNA-SEQ",
+                "filename": None,
+            },
+            experiment,
+        )
+        rnas.append(sample.accession_code)
+
+        dset = Dataset()
+        dset.data = {"GSE1234": micros, "GSE5678": rnas}
+        dset.scale_by = "NONE"
+        dset.aggregate_by = "SPECIES"
+        dset.svd_algorithm = "ARPACK"
+        dset.quantile_normalize = True
+        dset.save()
+
+        pjda = ProcessorJobDatasetAssociation()
+        pjda.processor_job = job
+        pjda.dataset = dset
+        pjda.save()
+
+        imputation_index = create_compendia.COMPENDIA_PIPELINE.index(
+            create_compendia._perform_imputation
+        )
+
+        pipeline = Pipeline(name=PipelineEnum.CREATE_COMPENDIA.value)
+        job_context = utils.run_pipeline(
+            {"job_id": job.id, "pipeline": pipeline},
+            create_compendia.COMPENDIA_PIPELINE[:imputation_index],
+        )
+
+        # First, run the imputation step without removing anything to get a baseline
+        expected_context = utils.run_pipeline(
+            job_context.copy(), [create_compendia.COMPENDIA_PIPELINE[imputation_index]]
+        )
+
+        # Now pick some rows to remove according to the instructions from
+        # https://github.com/AlexsLemonade/refinebio/pull/2879#issuecomment-895143336
+
+        random.seed(42)
+
+        # Select some rows randomly and mask ~30% of the values
+        rare_rows = random.sample(list(job_context["microarray_matrix"].index), k=25)
+        rare_genes = {}
+        for row in rare_rows:
+            cols = random.sample(list(job_context["microarray_matrix"].columns), k=110)
+            rare_genes[row] = cols
+            for col in cols:
+                job_context["microarray_matrix"].loc[row, col] = np.nan
+
+        # Now some entries from the other rows to mask completely at random
```

Comment:
```suggestion
        # Now some entries from the other rows to completely mask at random
```
I originally thought this meant "completely at random" until I reread Jackie's message.